### DISCUSSION TITLE
Title: Need Help Resolving DataTypeNotSupportException in Milvus

### DISCUSSION THREAD

**Discussion by GermanKevik** [milvus-io/milvus#32547] [Q&A and General discussion] [answered]
Hello Milvus Community,

I am currently experiencing a persistent issue with inserting data into a Milvus collection and am reaching out for assistance. Despite following the documentation and ensuring that my data matches the expected format and types for the schema, I continually encounter a DataTypeNotSupportException.

Here is a breakdown of the problem:
Problem Description

Whenever I attempt to insert data into a Milvus partition using the following data and schema definitions, I receive an error indicating that the data type is not supported.
Schema Setup

I have set up my schema as follows:

```
from pymilvus import FieldSchema, CollectionSchema, DataType, connections

id_field = FieldSchema(name="id", dtype=DataType.INT64, is_primary=True)
vector_field = FieldSchema(name="text_vector", dtype=DataType.FLOAT_VECTOR, dim=768)

schema = CollectionSchema(fields=[id_field, vector_field])
connections.connect(host=MILVUS_SETTINGS['host'], port=MILVUS_SETTINGS['port'])

milvus_collection = Collection(name="test_collection_name", schema=schema)
milvus_partition=Partition(milvus_collection, "test_partition_name")

unique_id = 1
text_vector = [0.1] * 768  # Make sure this has exactly 768 floats

###### Option 1 #######
data = {
    "id": [unique_id, unique_id + 1],  # List of IDs
    "text_vector": [text_vector, text_vector]  # List of vectors
}
milvus_partition.insert(data=data)
 --> Error during insert: <DataTypeNotSupportException: (code=0, message=Data type is not support.)>

###### Option 2 #######
data = [{
    "id": [unique_id, unique_id + 1],  # List of IDs
    "text_vector": [text_vector, text_vector]  # List of vectors
}]
milvus_partition.insert(data=data)
---> /.local/lib/python3.10/site-packages/pymilvus/orm/prepare.py", line 60, in prepare_insert_data
    if isinstance(data[i], numpy.ndarray):
IndexError: list index out of range
```

Seeking Community Insights

Has anyone encountered similar issues, or does anyone see a potential mistake in my setup? Any insights, suggestions, or guidance would be greatly appreciated as I look to resolve this problem.

Thank you all for your support!

**Comment by xiaofan-luan** [comment]
text_vector is not in the right format, it should be a python list

data=[
        [0.19886812562848388, 0.06023560599112088, 0.6976963061752597, 0.2614474506242501, 0.838729485096104],
        [0.3172005263489739, 0.9719044792798428, -0.36981146090600725, -0.4860894583077995, 0.95791889146345]
    ],

I'm not familiar with python but text_vector = [0.1] * 768 seems not be able to give you a 768 dimension vectors.
 Better to directly get vector from real emebedding model or using langchain

**Reply by GermanKevik** [reply]
Thank you for your help. The data was indeed wrong formated.

**Comment by yhmo** [comment, marked as answer]
```
    id_field = FieldSchema(name="id", dtype=DataType.INT64, is_primary=True)
    vector_field = FieldSchema(name="text_vector", dtype=DataType.FLOAT_VECTOR, dim=768)

    schema = CollectionSchema(fields=[id_field, vector_field])
    # connections.connect(host=MILVUS_SETTINGS['host'], port=MILVUS_SETTINGS['port'])

    milvus_collection = Collection(name="test_collection_name", schema=schema)
    milvus_partition = Partition(milvus_collection, "test_partition_name")

    unique_id = 1
    text_vector = [0.1] * 768  # Make sure this has exactly 768 floats

    ###### Option 1 #######
    data = [
        [unique_id, unique_id + 1],  # List of IDs
        [text_vector, text_vector]  # List of vectors
    ]
    milvus_partition.insert(data=data)
```

**Reply by GermanKevik** [reply]
Thank you, that was it.
